### ISSUE TITLE
Could not pass event PlayerInteractEvent/BlockDamageEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080

### ISSUE THREAD

**Issue by L4BORG** [NoCheatPlus/Issues#411]
#### Short description of the issue.
Exception spamming console every now and then, idk how to replicate it yet.

#### Complete output of the _ncp version_ command (please prefer text here).
```
>ncp version
[10:28:53 INFO]: ---- Version information ----
[10:28:53 INFO]: #### Server ####
[10:28:53 INFO]: git-TacoSpigot-"65fd35f" ~MC: 1.8.8~
[10:28:53 INFO]:   detected: 1.8.8
[10:28:53 INFO]: #### NoCheatPlus ####
[10:28:53 INFO]: Plugin: 3.15.0-SNAPSHOT-sMD5NET-b1080
[10:28:53 INFO]: MCAccess: 1.8.4-1.8.8 / Spigot-CB-1.8_R3
[10:28:53 INFO]: Features:
[10:28:53 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8
[10:28:53 INFO]:   checks: FastConsume | Gutenberg | AttackFrequency | FlyingFrequency | KeepAliveFrequency
[10:28:53 INFO]:   defaults: pvpKnockBackVelocity
[10:28:53 INFO]:   packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance | CatchAllAdapter
[10:28:53 INFO]: #### Related Plugins ####
[10:28:53 INFO]: ProtocolLib v3.6.5
```

#### Have you made any (noteworthy) changes to the default configuration of NCP?
No.

#### Do you run any other special plugins that affect game mechanics next to NC+ (adding skills, machines, adding/changing blocks, other anti cheating)?
ViaVersion.

#### Provide additional information, such as error/debug/violation logs. (For reproducible issues with moving consider: https://github.com/NoCheatPlus/Docs/wiki/Debugging#on-the-fly-debug-output-for-individual-players) Use GitHub gists or http://pastebin.com/ or similar for huge logs, or attach a (zip-) file here. For image uploads, please prefer a site that allows viewing without JavaScript, you can upload file attachments for issues right here.

```[06:51:07] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:228) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:195) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:191) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1232) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInArmAnimation.a(PacketPlayInArmAnimation.java:18) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInArmAnimation.a(PacketPlayInArmAnimation.java:1) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:419) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:408) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.blockinteract.BlockInteractListener.onPlayerInteract(BlockInteractListener.java:92) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor206.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	... 19 more
[06:51:07] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:228) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:195) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerInteractManager.a(PlayerInteractManager.java:107) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:626) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockDig.a(SourceFile:40) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockDig.a(SourceFile:10) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:419) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:408) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.blockinteract.BlockInteractListener.onPlayerInteract(BlockInteractListener.java:92) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor206.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	... 19 more
[06:51:07] [Server thread/ERROR]: Could not pass event BlockDamageEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callBlockDamageEvent(CraftEventFactory.java:264) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerInteractManager.a(PlayerInteractManager.java:175) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:626) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockDig.a(SourceFile:40) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockDig.a(SourceFile:10) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:419) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:408) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.blockbreak.BlockBreakListener.onBlockDamageLowest(BlockBreakListener.java:229) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor320.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	... 18 more
```

**Comment by L4BORG** [comment]
Actually there's more....

```[07:01:20] [Server thread/ERROR]: Could not pass event EntityDamageByEntityEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callEvent(CraftEventFactory.java:87) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callEntityDamageEvent(CraftEventFactory.java:553) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.handleEntityDamageEvent(CraftEventFactory.java:466) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.handleLivingEntityDamageEvent(CraftEventFactory.java:585) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityLiving.d(EntityLiving.java:1112) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityHuman.d(EntityHuman.java:860) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityLiving.damageEntity(EntityLiving.java:748) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityHuman.damageEntity(EntityHuman.java:801) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityPlayer.damageEntity(EntityPlayer.java:504) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityHuman.attack(EntityHuman.java:1002) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityPlayer.attack(EntityPlayer.java:1086) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1387) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInUseEntity.a(SourceFile:52) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInUseEntity.a(SourceFile:11) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException```

```[07:06:21] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:228) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerInteractManager.interact(PlayerInteractManager.java:463) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:763) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:52) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:1) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:419) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:408) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.blockinteract.BlockInteractListener.onPlayerInteract(BlockInteractListener.java:92) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor206.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	... 18 more
[07:06:21] [Server thread/ERROR]: Could not pass event BlockPlaceEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callBlockPlaceEvent(CraftEventFactory.java:127) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.ItemStack.placeItem(ItemStack.java:160) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerInteractManager.interact(PlayerInteractManager.java:503) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:763) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:52) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:1) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException```

**Comment by L4BORG** [comment]
```
[08:14:14] [Server thread/ERROR]: Could not pass event InventoryOpenEvent to NoCheatPlus v3.15.0-SNAPSHOT-sMD5NET-b1080
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:302) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:729) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.craftbukkit.v1_8_R3.event.CraftEventFactory.callInventoryOpenEvent(CraftEventFactory.java:715) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityPlayer.openTrade(EntityPlayer.java:717) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.EntityVillager.a(EntityVillager.java:136) [patched.jar:git-TacoSpigot-"65fd35f"]
	at sun.reflect.GeneratedMethodAccessor383.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at es.minetsii.eggwars.utils.villagerGui.NMSMerchant.openTrading(NMSMerchant.java:115) [EggWars.jar:?]
	at es.minetsii.eggwars.utils.villagerGui.Merchant.openTrading(Merchant.java:78) [EggWars.jar:?]
	at es.minetsii.eggwars.listeners.ClickInventoryListener.click(ClickInventoryListener.java:35) [EggWars.jar:?]
	at sun.reflect.GeneratedMethodAccessor269.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) [patched.jar:git-TacoSpigot-"65fd35f"]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:74) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:501) [patched.jar:git-TacoSpigot-"65fd35f"]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:486) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnection.a(PlayerConnection.java:1662) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(SourceFile:31) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PacketPlayInWindowClick.a(SourceFile:9) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.PlayerConnectionUtils$1.run(SourceFile:13) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_121]
	at net.minecraft.server.v1_8_R3.SystemUtils.a(SourceFile:44) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.B(MinecraftServer.java:774) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.DedicatedServer.B(DedicatedServer.java:378) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.A(MinecraftServer.java:713) [patched.jar:git-TacoSpigot-"65fd35f"]
	at net.minecraft.server.v1_8_R3.MinecraftServer.run(MinecraftServer.java:616) [patched.jar:git-TacoSpigot-"65fd35f"]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
Caused by: java.lang.NullPointerException
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:419) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.moving.util.MovingUtil.hasScheduledPlayerSetBack(MovingUtil.java:408) ~[?:?]
	at fr.neatmonster.nocheatplus.checks.inventory.InventoryListener.onPlayerInventoryOpen(InventoryListener.java:402) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor273.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:300) ~[patched.jar:git-TacoSpigot-"65fd35f"]
	... 34 more
```

**Comment by L4BORG** [comment]
https://github.com/NoCheatPlus/NoCheatPlus/blob/4b9e7c9fe311f6019e342af958b58c965c710303/NCPCore/src/main/java/fr/neatmonster/nocheatplus/checks/moving/util/MovingUtil.java#L419

DataManager is returning null for some reason?

**Comment by asofold** [comment]
Thank you for the fast reaction. The method used here can return null, if no PlayerData instance is there. This is intended behavior - the MovingUtil.hasScheduledPlayerSetBack did not check if it's null.

The PlayerData mechanics isn't fully implemented yet, so it's still possible to remove it for online players, though i would assume it's present after login. This will be subject to a more intense overhaul, because i intend to put all the check data into PlayerData, likely PlayerData will also store some shared data directly (like the last world) for the fastest thread-safe checking. Then online players will always have a PlayerData instance, though i still have to look out for NPC plugins (also combat tagging) creating Player instances that are not online, as well as reloading and plugin managers (so players haven't joined but are online and the data has to be created lazily).

**Comment by L4BORG** [comment]
It looks like this was fixed in #1081
